buildrunner/docker/runner.py

```diff
@@ -251,6 +254,94 @@ def cleanup(self):
 
         self.container = None
 
+    def restore_caches(self, caches: OrderedDict, cache_archive_ext: str):  # pylint: disable=too-many-branches,too-many-locals
+        """
+        Restores caches from the host system to the destination location in the docker container.
+        """
+        if caches is None or not isinstance(caches, OrderedDict):
+            raise TypeError(f"Caches should be type OrderDict instead of {type(caches)}")
+
+        restored_cache_src = set()
+        for local_cache_archive_file, docker_path in caches.items():
+            if docker_path in restored_cache_src:
+                print(f"Destination path {docker_path} has already been matched and restored to the container "
+                      f"skipping {local_cache_archive_file} -> {docker_path}")
+                continue
+
+            # Check for prefix matching
+            if not os.path.exists(local_cache_archive_file):
+                cache_dir = os.path.dirname(local_cache_archive_file)
+
+                if not os.path.exists(cache_dir):
+                    print(f"Cache directory {cache_dir} does not exist, "
+                          f"skipping restore of archive {local_cache_archive_file}")
+                    continue
```

Comment:
Yes it is done here -> https://github.com/adobe/buildrunner/blob/main/buildrunner/__init__.py#L578-L579

However, on at least one occasion I had commented out that lines which created the cached dir and the code in `restore_caches` threw an exception, since the directory didn't exist. Even though in theory the cache directory should always be created at this point in the code, I left the check in there anyways in case things change in the future. I can change it to print a warning or remove this check altogether and let it raise an exception if there is an issue.